beacon-chain/state/state-native/v1/getters_state.go

```diff
@@ -1,115 +1,116 @@
-func (b *BeaconState) InnerStateUnsafe() interface{} {
+func (b *BeaconState) ToProtoUnsafe() interface{} {
 	if b == nil {
 		return nil
 	}
-	return b.state
+
+	return &ethpb.BeaconState{
+		GenesisTime:                 b.genesisTime,
+		GenesisValidatorsRoot:       b.genesisValidatorsRoot[:],
+		Slot:                        b.slot,
+		Fork:                        b.fork,
+		LatestBlockHeader:           b.latestBlockHeader,
+		BlockRoots:                  state_native.BlockRootsToSlice(b.blockRoots),
+		StateRoots:                  state_native.StateRootsToSlice(b.stateRoots),
+		HistoricalRoots:             state_native.HistoricalRootsToSlice(b.historicalRoots),
+		Eth1Data:                    b.eth1Data,
+		Eth1DataVotes:               b.eth1DataVotes,
+		Eth1DepositIndex:            b.eth1DepositIndex,
+		Validators:                  b.validators,
+		Balances:                    b.balances,
+		RandaoMixes:                 state_native.RandaoMixesToSlice(b.randaoMixes),
+		Slashings:                   b.slashings,
+		PreviousEpochAttestations:   b.previousEpochAttestations,
+		CurrentEpochAttestations:    b.currentEpochAttestations,
+		JustificationBits:           b.justificationBits,
+		PreviousJustifiedCheckpoint: b.previousJustifiedCheckpoint,
+		CurrentJustifiedCheckpoint:  b.currentJustifiedCheckpoint,
+		FinalizedCheckpoint:         b.finalizedCheckpoint,
+	}
 }
 
-// CloneInnerState the beacon state into a protobuf for usage.
-func (b *BeaconState) CloneInnerState() interface{} {
-	if b == nil || b.state == nil {
+// ToProto the beacon state into a protobuf for usage.
+func (b *BeaconState) ToProto() interface{} {
+	if b == nil {
 		return nil
 	}
 
 	b.lock.RLock()
 	defer b.lock.RUnlock()
+
 	return &ethpb.BeaconState{
-		GenesisTime:                 b.genesisTime(),
-		GenesisValidatorsRoot:       b.genesisValidatorRoot(),
-		Slot:                        b.slot(),
-		Fork:                        b.fork(),
-		LatestBlockHeader:           b.latestBlockHeader(),
-		BlockRoots:                  b.blockRoots(),
-		StateRoots:                  b.stateRoots(),
-		HistoricalRoots:             b.historicalRoots(),
-		Eth1Data:                    b.eth1Data(),
-		Eth1DataVotes:               b.eth1DataVotes(),
-		Eth1DepositIndex:            b.eth1DepositIndex(),
-		Validators:                  b.validators(),
-		Balances:                    b.balances(),
-		RandaoMixes:                 b.randaoMixes(),
-		Slashings:                   b.slashings(),
-		PreviousEpochAttestations:   b.previousEpochAttestations(),
-		CurrentEpochAttestations:    b.currentEpochAttestations(),
-		JustificationBits:           b.justificationBits(),
-		PreviousJustifiedCheckpoint: b.previousJustifiedCheckpoint(),
-		CurrentJustifiedCheckpoint:  b.currentJustifiedCheckpoint(),
-		FinalizedCheckpoint:         b.finalizedCheckpoint(),
+		GenesisTime:                 b.genesisTime,
+		GenesisValidatorsRoot:       b.genesisValidatorsRoot[:],
```

Comment:
You should copy the whole root, instead of taking its slice. Otherwise there is an incomplete copy here, where the slice references the genesis root from the old state.